modules/core/04-channel/keeper/upgrade.go

```diff
@@ -278,8 +278,8 @@ func (k Keeper) validateUpgradeFields(ctx sdk.Context, proposedUpgrade types.Upg
 	}
 
 	connectionID := proposedUpgrade.ConnectionHops[0]
-	connection, err := k.GetConnection(ctx, connectionID)
-	if err != nil {
+	connection, found := k.connectionKeeper.GetConnection(ctx, connectionID)
```

Comment:
This is because `k.GetConnection` was returning the interface `exported.ConnectionI` rather than the concrete `ConnectionEnd` type, right? So that `GetVersions()` can be used below. 

Should we add `GetVersions()` as an interface method to `exported.ConnectionI`? cc. @colin-axner